services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedDeviceBackend.java

```diff
@@ -85,22 +90,52 @@ public FileBasedDeviceBackend(
         this.tenantInformationService = tenantInformationService;
     }
 
+    /**
+     * Sets the service to use for auto-provisioning devices and gateways.
+     * <p>
+     * If the service is not configured, auto-provisioning will be disabled.
+     *
+     * @param deviceAndGatewayAutoProvisioner The service to use for auto-provisioning devices and gateways.
+     * @throws NullPointerException if the service is {@code null}.
+     */
+    public void setDeviceAndGatewayAutoProvisioner(
+            final DeviceAndGatewayAutoProvisioner deviceAndGatewayAutoProvisioner) {
+        this.deviceAndGatewayAutoProvisioner = Objects.requireNonNull(deviceAndGatewayAutoProvisioner);
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public Future<Void> start() {
+        final List<Future> services = new ArrayList<>();
+
         LOG.debug("starting up services");
-        return CompositeFuture.all(registrationService.start(), credentialsService.start()).mapEmpty();
+        services.add(registrationService.start());
+        services.add(credentialsService.start());
+        Optional.ofNullable(deviceAndGatewayAutoProvisioner)
+                .map(AbstractAutoProvisioningEventSender::start)
+                .map(services::add);
+
+        return CompositeFuture.all(services)
+                .mapEmpty();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public Future<Void> stop() {
+        final List<Future> services = new ArrayList<>();
+
         LOG.debug("stopping services");
-        return CompositeFuture.join(registrationService.stop(), credentialsService.stop()).mapEmpty();
+        services.add(registrationService.start());
+        services.add(credentialsService.start());
+        Optional.ofNullable(deviceAndGatewayAutoProvisioner)
+                .map(AbstractAutoProvisioningEventSender::start)
```

Comment:
DeviceAndGatewayAutoProvisioner::stop?

Reply:
I have pushed the changes for this and for all the above comments.